src/audio/pipeline/pipeline-stream.c

```diff
@@ -502,8 +502,16 @@ void pipeline_get_timestamp(struct pipeline *p, struct comp_dev *host,
 	data.posn = posn;
 
 	if (walk_ctx.comp_func(host, NULL, &walk_ctx, host->direction) !=
-	    PPL_STATUS_PATH_STOP)
-		pipe_dbg(p, "pipeline_get_timestamp(): DAI position update failed");
+	    PPL_STATUS_PATH_STOP) {
+		struct comp_dev *dai = pipeline_get_dai_comp(host->pipeline->pipeline_id);
+
+		if (!dai) {
+			pipe_dbg(p, "pipeline_get_timestamp(): DAI position update failed");
+			return;
+		}
+
+		platform_dai_timestamp(dai, posn);
```

Comment:
but do we need the `pipeline_comp_timestamp()` recursion at all then? Wouldn't it be enough to just do this directly instead?

Reply:
@lyakh for us platform_dai_timestamp() would be enough but I wonder what was the initial intention of this code. 

Does DAI timestamps feature work on Intel platforms? 

It looks like we cannot next pipeline_copy() with pipeline_get_timestamp(). But this happens on i.MX because we have a "software" Host DMA, so there is no async dma callback.

Reply:
At least this version doesn't seem to break CI. I've just had a look at the `pipeline_get_dai_comp()` function and it's only so far used in one place in the code on an IPC4 path. Looking at the function it seems to only handle the playback direction?

Reply:
That seems to be the case, yes.